diff --git a/src/test/java/org/apache/commons/cli/AmbiguousOptionExceptionTest.java b/src/test/java/org/apache/commons/cli/AmbiguousOptionExceptionTest.java
new file mode 100644
index 00000000..219a6d7b
--- /dev/null
+++ b/src/test/java/org/apache/commons/cli/AmbiguousOptionExceptionTest.java
@@ -0,0 +1,63 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package org.apache.commons.cli;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@link AmbiguousOptionException}.
+ */
+class AmbiguousOptionExceptionTest {
+
+    @Test
+    void testConstructor() {
+        final Collection<String> matchingOptions = Arrays.asList("optA", "optB");
+        final AmbiguousOptionException exception = new AmbiguousOptionException("o", matchingOptions);
+        assertEquals("Ambiguous option: 'o'  (could be: 'optA', 'optB')", exception.getMessage());
+        assertEquals("o", exception.getOption());
+    }
+
+    @Test
+    void testGetMatchingOptions() {
+        final Collection<String> matchingOptions = Arrays.asList("optA", "optB");
+        final AmbiguousOptionException exception = new AmbiguousOptionException("o", matchingOptions);
+        assertEquals(Arrays.asList("optA", "optB"), exception.getMatchingOptions());
+    }
+
+    @Test
+    void testSingleMatchingOption() {
+        final Collection<String> matchingOptions = Collections.singletonList("optA");
+        final AmbiguousOptionException exception = new AmbiguousOptionException("o", matchingOptions);
+        assertEquals("Ambiguous option: 'o'  (could be: 'optA')", exception.getMessage());
+    }
+
+    @Test
+    void testEmptyMatchingOptions() {
+        final Collection<String> matchingOptions = new ArrayList<>();
+        final AmbiguousOptionException exception = new AmbiguousOptionException("o", matchingOptions);
+        assertEquals("Ambiguous option: 'o'  (could be: )", exception.getMessage());
+        assertEquals("o", exception.getOption());
+    }
+}
\ No newline at end of file
