diff --git a/src/main/java/org/apache/commons/cli/AlreadySelectedException.java b/src/main/java/org/apache/commons/cli/AlreadySelectedException.java
index 7603eac5..70564c7d 100644
--- a/src/main/java/org/apache/commons/cli/AlreadySelectedException.java
+++ b/src/main/java/org/apache/commons/cli/AlreadySelectedException.java
@@ -28,7 +28,7 @@ public class AlreadySelectedException extends ParseException {
     private static final long serialVersionUID = 3674381532418544760L;
 
     /** The option group selected. */
-    private final OptionGroup optionGroup;
+    private final OptionGroup group;
 
     /** The option that triggered the exception. */
     private final Option option;
@@ -36,13 +36,13 @@ public class AlreadySelectedException extends ParseException {
     /**
      * Constructs a new {@code AlreadySelectedException} for the specified option group.
      *
-     * @param optionGroup the option group already selected.
+     * @param group the option group already selected.
      * @param option the option that triggered the exception.
      * @since 1.2
      */
-    public AlreadySelectedException(final OptionGroup optionGroup, final Option option) {
+    public AlreadySelectedException(final OptionGroup group, final Option option) {
         this(String.format("The option '%s' was specified but an option from this group has already been selected: '%s'", option.getKey(),
-                optionGroup.getSelected()), optionGroup, option);
+                group.getSelected()), group, option);
     }
 
     /**
@@ -54,9 +54,9 @@ public class AlreadySelectedException extends ParseException {
         this(message, null, null);
     }
 
-    private AlreadySelectedException(final String message, final OptionGroup optionGroup, final Option option) {
+    private AlreadySelectedException(final String message, final OptionGroup group, final Option option) {
         super(message);
-        this.optionGroup = optionGroup;
+        this.group = group;
         this.option = option;
     }
 
@@ -77,6 +77,6 @@ public class AlreadySelectedException extends ParseException {
      * @since 1.2
      */
     public OptionGroup getOptionGroup() {
-        return optionGroup;
+        return group;
     }
 }
diff --git a/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java b/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
index 86cefd19..6e71a15f 100644
--- a/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
+++ b/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
@@ -40,20 +40,20 @@ public class AmbiguousOptionException extends UnrecognizedOptionException {
      * @return
      */
     private static String createMessage(final String option, final Collection<String> matchingOptions) {
-        final StringBuilder buf = new StringBuilder("Ambiguous option: '");
-        buf.append(option);
-        buf.append("'  (could be: ");
-        final Iterator<String> it = matchingOptions.iterator();
-        while (it.hasNext()) {
-            buf.append(Char.APOS);
-            buf.append(it.next());
-            buf.append(Char.APOS);
-            if (it.hasNext()) {
-                buf.append(", ");
+        final StringBuilder builder = new StringBuilder("Ambiguous option: '");
+        builder.append(option);
+        builder.append("'  (could be: ");
+        final Iterator<String> iterator = matchingOptions.iterator();
+        while (iterator.hasNext()) {
+            builder.append(Char.APOS);
+            builder.append(iterator.next());
+            builder.append(Char.APOS);
+            if (iterator.hasNext()) {
+                builder.append(", ");
             }
         }
-        buf.append(")");
-        return buf.toString();
+        builder.append(")");
+        return builder.toString();
     }
 
     /** The list of options matching the partial name specified */
