diff --git a/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java b/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
index 86cefd19..6e71a15f 100644
--- a/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
+++ b/src/main/java/org/apache/commons/cli/AmbiguousOptionException.java
@@ -40,20 +40,20 @@ public class AmbiguousOptionException extends UnrecognizedOptionException {
      * @return
      */
     private static String createMessage(final String option, final Collection<String> matchingOptions) {
-        final StringBuilder buf = new StringBuilder("Ambiguous option: '");
-        buf.append(option);
-        buf.append("'  (could be: ");
-        final Iterator<String> it = matchingOptions.iterator();
-        while (it.hasNext()) {
-            buf.append(Char.APOS);
-            buf.append(it.next());
-            buf.append(Char.APOS);
-            if (it.hasNext()) {
-                buf.append(", ");
+        final StringBuilder builder = new StringBuilder("Ambiguous option: '");
+        builder.append(option);
+        builder.append("'  (could be: ");
+        final Iterator<String> iterator = matchingOptions.iterator();
+        while (iterator.hasNext()) {
+            builder.append(Char.APOS);
+            builder.append(iterator.next());
+            builder.append(Char.APOS);
+            if (iterator.hasNext()) {
+                builder.append(", ");
             }
         }
-        buf.append(")");
-        return buf.toString();
+        builder.append(")");
+        return builder.toString();
     }
 
     /** The list of options matching the partial name specified */
